### PR TITLE
CVE search results clean-up

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Fixed remove filter bug
- Refactored stale JS
- Reinstated package query functionality
- Deleted stale templates
- Added no results view

## QA

- View the site locally in your web browser at: https://ubuntu-com-13893.demos.haus/security/cves?q=
- Test filtering functionality
- Test querying by affected package
- Test querying by invalid cve and see that the no results notification looks as expected
- See that all works as expected

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-11310, https://warthogs.atlassian.net/browse/WD-8041

### CODE REVIEW

**Comment by juanruitina** [comment]
- [ ] Status filtering doesn't seem to be working: when I select "Fixed", the first result doesn't have any status as fixed/released: https://ubuntu-com-13893.demos.haus/security/cves?q=&version=noble&status=released
- [ ] Let's please make sure LTS releases show up as such (e.g. 24.04 LTS).
- [ ] For `.cve-filters .p-side-navigation .release-accordion-group`, let's change `overflow: scroll` with `overflow-y: auto` so we prevent unnecessary scrollbars.

**Comment by juanruitina** [comment]
For the LTS ones, let's make it look like the design: "24.04 LTS (black) noble (dark gray)":

Screenshot 2024-06-03 at 16 21 00

File: templates/security/cve/_cve-landing.html [hunk 1]
```diff
@@ -1,32 +1,44 @@
 <section class="p-section">
```
Comment (mtruj013) [inline templates/security/cve/_cve-landing.html:1]: No real changes here, only linting

File: templates/security/cve/_cve-search-results.html [hunk 2]
```diff
@@ -44,6 +44,20 @@
...
+        </div>
+        <div class="p-notification--information">
+          <div class="p-notification__content">
+            <p class="p-notification__title p-heading--5">There is no CVE that matches that search.</p>
+            <p class="p-notification__message">
+              Try another search, <a href="/security/cves?q=">clear all filters</a>, or <a href="/security/cves?q=">see the full list of CVEs</a>
+            </p>
+          </div>
+        </div>
+      </div>
```
Comment (mtruj013) [inline templates/security/cve/_cve-search-results.html:60]: Change here, the rest is linting

File: templates/security/cve/_cve-search.html [hunk 3]
```diff
@@ -2,13 +2,12 @@
   <div class="row">
```
Comment (mtruj013) [inline templates/security/cve/_cve-search.html:2]: Linting

File: templates/security/cve/filters/release-filter.html [hunk 4]
```diff
@@ -1,29 +1,40 @@
...
-  <a href="" class="js-show-unmaintained-releases">Show unmaintained releases</a>
-  <div class="js-unmaintained-releases u-hide">
-    <p class="p-text--small-caps">Unmaintained</p>
-    {% for release in unmaintained_releases %}
-      <label class="p-checkbox">
-        <input type="checkbox" aria-labelledby={{release.version}} class="p-checkbox__input" value={{release.codename}} />
-        <span class="p-checkbox__label" id={{release.version}}>{{release.version}} <span class="u-text--muted">{{release.codename}}</span></span>
+      <input type="checkbox" aria-labelledby={{ release.version }} class="p-checkbox__input" value={{ release.codename }} />
+      <span class="p-checkbox__label" id={{ release.version }}>{{ release.version }}
+        {% if release.support_tag %}{{ release.support_tag }}{% endif %}
```
Comment (mtruj013) [inline templates/security/cve/filters/release-filter.html:13]: Change here

File: templates/security/cve/filters/release-filter.html [hunk 5]
```diff
@@ -1,29 +1,40 @@
...
+    <p class="p-text--small-caps">
+      Still maintained with
+      <br />
+      Ubuntu Pro
+    </p>
+    {% for release in lts_releases %}
+      <label class="p-checkbox">
+        <input type="checkbox" aria-labelledby={{ release.version }} class="p-checkbox__input" value={{ release.codename }} />
+        <span class="p-checkbox__label" id={{ release.version }}>{{ release.version }}
+          {% if release.lts == True %}LTS{% endif %}
```
Comment (mtruj013) [inline templates/security/cve/filters/release-filter.html:26]: Change here

**Comment by juanruitina** [comment]
UX+1, let's fix the API issue whenever we can.

File: static/js/src/cve/cve.js [hunk 6]
```diff
@@ -44,74 +36,22 @@ const ltsReleases = Object.values(ltsReleasesObj).map(
...
+// to URLSearchParams object
+function handlePackageInput() {
+  packageInput.addEventListener("input", function (event) {
+    urlParams.set("package", packageInput.value);
+  });
+}
+handlePackageInput();
 
 // Adds event listeners to all filter checkboxes
 function handleFilters() {
```
Comment (britneywwc) [inline static/js/src/cve/cve.js:57]: Could it be helpful to add this check before line 60 to tick all the maintained releases' checkboxes when **All maintained releases** is selected? This way we can reduce the need to loop through the array of maintained releases as we add/remove params.

```
if (checkbox.value === "maintained") {
  maintainedReleases.forEach(function (release) {
    const checkbox = getCheckboxFromRelease(release);
    checkbox?.checked = event.target.checked;
  });
}
```

```
function getCheckboxFromRelease(release) {
  const releaseCheckboxesArray = Array.from(releaseCheckboxes);
  const checkbox = releaseCheckboxesArray.find(checkbox => checkbox.value === release);

  return checkbox;
}
```

File: static/js/src/cve/cve.js [hunk 7]
```diff
@@ -227,27 +179,27 @@ function handleClearFilters() {
...
 function addParam(param, value) {
   if (value === "maintained") {
     maintainedReleases.forEach(function (release) {
-      filterParams.push(
-        { param: param, value: release },
-        { param: "status", value: "" }
-      );
+      if (!urlParams.has(param, release)) {
+        urlParams.append(param, release);
+      }
```
Comment (britneywwc) [inline static/js/src/cve/cve.js:191]: We can remove this catch if the all checkboxes are already ticked before submit.

File: static/js/src/cve/cve.js [hunk 8]
```diff
@@ -148,19 +87,36 @@ function handleFilters() {
...
 function removeParam(param, value) {
-  if (param === "version") {
-    let statusIndex = filterParams
-      .map((param) => param.param)
-      .indexOf("status");
-    filterParams.splice(statusIndex, 1);
+  if (value === "maintained") {
+    maintainedReleases.forEach(function (release) {
+      urlParams.delete(param, release);
+    });
```
Comment (britneywwc) [inline static/js/src/cve/cve.js:95]: We can remove this catch if all the checkboxes are already unticked before submit.

File: static/js/src/cve/cve.js [hunk 9]
```diff
@@ -227,27 +179,27 @@ function handleClearFilters() {
...
+        urlParams.append(param, status);
+      }
     });
   } else {
-    filterParams.push({ param: param, value: value });
+    if (!urlParams.has(param, value)) {
+      urlParams.append(param, value);
+    }
   }
 }
```
Comment (britneywwc) [inline static/js/src/cve/cve.js:205]: There is a typo for `showUnmaintainedReleases()` on line 206 and 213.

**Review by britneywwc — CHANGES_REQUESTED** [review]
Thanks @mtruj013! I added some suggestions for the maintained releases checkboxes and there is a minor typo in the file.

I also found a bug where if you applied filters then performed a search query, the previous filter parameters get removed. The vice versa is not true though, when you perform a search query then filter, both parameters still stay. Maybe this is something worth investigating further.

File: static/js/src/cve/cve.js (hunk 6 above)
Comment (mtruj013) [reply static/js/src/cve/cve.js:71]: Great catch, thanks!

**Comment by mtruj013** [comment]
Ready for you @britneywwc !

**Review by britneywwc — APPROVED** [review]
Looks great, thanks! 🎉

**Pull request merged**